src/pkg/services/m365/api/config.go

```diff
@@ -0,0 +1,85 @@
+package api
+
+import (
+	"fmt"
+	"strings"
+
+	abstractions "github.com/microsoft/kiota-abstractions-go"
+)
+
+const (
+	maxNonDeltaPageSize = int32(999)
+	maxDeltaPageSize    = int32(500)
+)
+
+// selectable values, case insensitive
+// not comprehensive - just adding ones that might
+// get easily misspelled.
+// eg: we don't need a const for "id"
+const (
+	parentFolderID    = "parentFolderId"
+	displayName       = "displayName"
+	userPrincipalName = "userPrincipalName"
+)
+
+// header keys
+const (
+	headerKeyConsistencyLevel = "ConsistencyLevel"
+	headerKeyPrefer           = "Prefer"
+)
+
+// header values
+const (
+	idTypeImmutable = `IdType="ImmutableId"`
```

Comment:
should this also be split as a key and a value? Technically it can have different values, we were only interested in immutable IDs though those still have issues that need resolving

Reply:
Going to keep it as is.  Should be easy to update when we want to introduce new values.